lib/rack/server.rb

```diff
@@ -381,6 +381,7 @@ def pidfile_process_status
 
         pid = ::File.read(options[:pid]).to_i
         return :dead if pid == 0
+        return :this_process if pid == Process.pid
```

Comment:
It seems like possible bikeshedding, but `:this_process` is semantically-clear whereas `:self` conveys less information, more ambiguously.